ambry-clustermap/src/test/java/com.github.ambry.clustermap/ClusterMapUtilsTest.java

```diff
@@ -220,25 +238,41 @@ private void verifyWritablePartitionsReturned(ClusterMapUtils.PartitionSelection
       Set<MockPartitionId> expectedReturnForClassNotBeingTested) {
     Set<MockPartitionId> expectedReturnForClassBeingTested = new HashSet<>(Arrays.asList(testedPart1, testedPart2));
     // no problematic scenarios
-    assertCollectionEquals("Partitions returned not as expected", allPartitionIds, psh.getWritablePartitions(null));
+    assertCollectionEquals("Partitions returned not as expected", allPartitionIds,
+        psh.getWritablePartitions(null));
     assertCollectionEquals("Partitions returned not as expected", expectedReturnForClassBeingTested,
         psh.getWritablePartitions(classBeingTested));
+    assertInCollection("Random partition returned not as expected", allPartitionIds,
+        psh.getRandomWritablePartition(null, null));
+    assertInCollection("Random partition returned not as expected", expectedReturnForClassBeingTested,
+        psh.getRandomWritablePartition(classBeingTested, null));
+    assertNull(psh.getRandomWritablePartition(null, new ArrayList<>(allPartitionIds)));
     checkCaseInsensitivityForPartitionSelectionHelper(psh, false, classBeingTested, expectedReturnForClassBeingTested);
     // one replica of one partition of "classBeingTested" down
     ((MockReplicaId) testedPart1.getReplicaIds().get(0)).markReplicaDownStatus(true);
     allPartitionIds.remove(testedPart1);
     expectedReturnForClassBeingTested.remove(testedPart1);
-    assertCollectionEquals("Partitions returned not as expected", allPartitionIds, psh.getWritablePartitions(null));
+    assertCollectionEquals("Partitions returned not as expected", allPartitionIds,
+        psh.getWritablePartitions(null));
     assertCollectionEquals("Partitions returned not as expected", expectedReturnForClassBeingTested,
         psh.getWritablePartitions(classBeingTested));
+    assertInCollection("Partitions returned not as expected", allPartitionIds,
```

Comment:
The message should be `Random partition returned not as expected` if you want to make it consistent in this test.

Reply:
fixed.